frontend/src/Table.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { CSSProperties, ReactNode } from "react";
+import {
+    TABLE_BORDER_COLOR,
+    TABLE_FONT_COLOR,
+    TABLE_HEADER_COLOR,
+} from "./styles/colors";
+
+export interface ColHeaderData {
+    key: string | number;
+    content: ReactNode;
+    style?: CSSProperties;
+}
+
+export interface CellData {
+    key: string | number;
+    content: ReactNode;
+    style?: CSSProperties;
+}
+
+export interface RowData {
+    key: string | number;
+    header: ReactNode;
+    bodyCells: CellData[];
+    style?: CSSProperties;
+}
+
+export interface CornerHeaderData {
+    content: ReactNode;
+    style?: CSSProperties;
+}
+
+interface Props {
+    cornerHeader?: CornerHeaderData;
+    colHeaders: ColHeaderData[];
+    rows: RowData[];
+    style?: CSSProperties;
+}
+
+export default function Table({
```

Comment:
So, I pulled the trigger on making a custom table, because I didn't want to fight with Material UI anymore 😬 my plan is to replace the leaderboard and game reports table with this custom one in a separate PR. It will fix or make easier a few outstanding issues (#168, #129, #96)

Reply:
You rock, love the ownership <3

Reply:
❤️